collection.js

```diff
@@ -1,63 +1,60 @@
 var EventEmitter = require('events').EventEmitter;
+var url = require('url');
+var util = require('util');
+var rels = require('zetta-rels');
+var uuid = require('node-uuid');
 
 var StatsCollector = module.exports = function() {
-  this.emitter = new EventEmitter();
-  this.server = null;
-};
-
-StatsCollector.prototype.on = function() {
-  this.emitter.on.apply(this.emitter, arguments);
+  EventEmitter.call(this);
 };
+util.inherits(StatsCollector, EventEmitter);
 
 StatsCollector.prototype.collect = function() {
   return this._collect.bind(this);
 };
-StatsCollector.prototype._collect = function(server) {
+
+StatsCollector.prototype._collect = function(runtime) {
   var self = this;
-  this.server = server;
 
-  var servers = []; // keep a list of peers subscribed to, to filter duplicates
-  server.pubsub.subscribe('_peer/connect', function(e, msg) {
-    if (servers.indexOf(msg.peer.name) > -1) {
+  runtime.pubsub.subscribe('_peer/connect', function(ev, msg) {
+    var topic = 'query:' + uuid.v4() + '/where type is not missing';
```

Comment:
@kevinswiber Because the query is unique every time the peer connects i'm seeing the queries build up on the peer_socket. On the first connect only one "query/where type is not missing" on the first reconnect the peer get's two of the those queries with different uuids and so forth.

I doesn't produce any duplicate data on output of the collector because the peer will only subscribe to the device's stream once and emit it only once. However it does build the number of device queries sent to the peer.